### ISSUE TITLE
Sometimes the proram froze due to unexpected deadlocks

### ISSUE THREAD

**Issue by gileli121** [BiGilSoft/WindowTop#235]
Sometimes I still get a few reports that the app sometimes froze.
It reproduced last time on v5.12.3

This time I choose a new approach to fix it. 
I implemented something that will reduce the number of threads.. no need for threads when I don't do stuff in parallel... no reason to create a thread just to use after that locks because it will be just like serial execution + unnecessary risk of deadlocks bugs.

The solution was to create something like this (simplified examples):

```c#
public static bool IsQueueRunning { get; set; }
static readonly ConcurrentStack<Action> queue = new ConcurrentStack<Action>();
```

And

```c#
public static void RunInNonUiThread(Action actionToRun)
{ 
	queue.Push(actionToRun);
}
```

And finally
```c#
public static void ProcessQueue()
{

	if (!IsQueueRunning)
		return;

	if (queue.TryPop(out var action))
	{
		action.Invoke();
	}
}
```

While `ProcessQueue()` is executed all the time via a single background thread

So I hope to finally fix it in this test version:

[WindowTop 5.12.4-beta - setup.exe.zip](https://github.com/BiGilSoft/WindowTop/files/8865123/WindowTop.5.12.4-beta.-.setup.exe.zip)

**Comment by gileli121** [comment]
Found regressions with the click-through feature.
I also found mistakes/unsafe code... 

Need to push to the `queue` only under this condition `Thread.CurrentThread.ManagedThreadId != QueueThreadId`
Otherwise, it is fine to run it directly.

New beta:
[WindowTop 5.12.4-beta2 - setup.exe.zip](https://github.com/BiGilSoft/WindowTop/files/8880259/WindowTop.5.12.4-beta2.-.setup.exe.zip)

**Comment by gileli121** [comment]
Verified as fixed.
No issues found.
Released in
https://github.com/BiGilSoft/WindowTop/releases/tag/v5.12.4

**Comment by gileli121** [comment]
Found small regression as a result of this fix.
Fixed in
https://github.com/BiGilSoft/WindowTop/releases/tag/v5.12.5

**Comment by gileli121** [comment]
Fixed again in 5.19.2 and released
